File: March 21, 2026.js
function decodeQr(qrCode) {

  const n = qrCode.length;
  let newMatrix = qrCode;
  let result = "";
  let rotations = 0;

  while(rotations < 4) {

    const topLeft =
      newMatrix[0][0] === "1" &&
      newMatrix[0][1] === "1" &&
      newMatrix[1][0] === "1" &&
      newMatrix[1][1] === "1";

    const topRight =
      newMatrix[0][n - 2] === "1" &&
      newMatrix[0][n - 1] === "1" &&
      newMatrix[1][n - 2] === "1" &&
      newMatrix[1][n - 1] === "1";

    const bottomLeft =
      newMatrix[n - 2][0] === "1" &&
      newMatrix[n - 2][1] === "1" &&
      newMatrix[n - 1][0] === "1" &&
      newMatrix[n - 1][1] === "1";

    if (topLeft && topRight && bottomLeft) {
      break;
    }

    newMatrix = rotate90(newMatrix);
    rotations++;

  }
  
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {

      const inTopLeft = i < 2 && j < 2;
      const inTopRight = i < 2 && j >= n - 2;
      const inBottomLeft = i >= n - 2 && j < 2;

      if (!inTopLeft && !inTopRight && !inBottomLeft) {
        result += newMatrix[i][j];
      }
    }
  }

  return result;
}


function rotate90(matrix) {
  const n = matrix.length;

  const arr = matrix.map(row => row.split(""));

  let rotated = Array.from({ length: n }, () => Array(n).fill(""));

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      rotated[j][n - 1 - i] = arr[i][j];
    }
  }

  return rotated.map(row => row.join(""));
}

File: March 21, 2026.py
def decode_qr(qr_code):
    
    n = len(qr_code)
    new_matrix = qr_code[:]
    result = ""
    rotations = 0

    while rotations < 4:

        top_left = (
            new_matrix[0][0] == "1" and
            new_matrix[0][1] == "1" and
            new_matrix[1][0] == "1" and
            new_matrix[1][1] == "1"
        )

        top_right = (
            new_matrix[0][n - 2] == "1" and
            new_matrix[0][n - 1] == "1" and
            new_matrix[1][n - 2] == "1" and
            new_matrix[1][n - 1] == "1"
        )

        bottom_left = (
            new_matrix[n - 2][0] == "1" and
            new_matrix[n - 2][1] == "1" and
            new_matrix[n - 1][0] == "1" and
            new_matrix[n - 1][1] == "1"
        )

        if top_left and top_right and bottom_left:
            break

        new_matrix = rotate_90(new_matrix)
        rotations += 1

    for i in range(n):
        for j in range(n):

            in_top_left = i < 2 and j < 2
            in_top_right = i < 2 and j >= n - 2
            in_bottom_left = i >= n - 2 and j < 2

            if not (in_top_left or in_top_right or in_bottom_left):
                result += new_matrix[i][j]

    return result


def rotate_90(matrix):
    n = len(matrix)

    arr = [list(row) for row in matrix]

    rotated = [[""] * n for _ in range(n)]

    for i in range(n):
        for j in range(n):
            rotated[j][n - 1 - i] = arr[i][j]

    return ["".join(row) for row in rotated]